File: app/src/main/java/com/library/testmvp/mvp/LoginContract.java
package com.library.testmvp.mvp;

import com.common.base.mvp.BaseView;
import com.common.http.base.BaseResponse;
import com.common.http.bean.ExampleBean;

import java.util.HashMap;

import io.reactivex.Observable;

/**
 * @Author: 张鹏飞
 * @Email: 1271396448@qq.com
 * <p>
 * @Desc:
 */

public interface LoginContract {

    interface Model {
        Observable<BaseResponse<ExampleBean>> login(HashMap<String, Object> map);
    }

    interface View extends BaseView {
        void loginSuccess(ExampleBean bean);
    }

    interface Presenter {
        void login(HashMap map);
    }
}

File: app/src/main/java/com/library/testmvp/mvp/LoginContract.kt
package com.library.testmvp.mvp

import com.common.base.mvp.BaseView
import com.common.http.base.BaseResponse
import com.common.http.bean.ExampleBean
import io.reactivex.Observable
import java.util.*

/**
 * @Author: 张鹏飞
 * @Email: 1271396448@qq.com
 *
 *
 * @Desc:
 */
interface LoginContract {
    interface Model {
        fun login(map: HashMap<String, ExampleBean>): Observable<BaseResponse<ExampleBean>>
    }

    interface View : BaseView {
        fun loginSuccess(bean: ExampleBean?)
    }

    interface Presenter {
        fun login(map: HashMap<String, ExampleBean>)
    }
}